acurl/acurl/__init__.py

```diff
@@ -379,7 +379,13 @@ def json(self):
     @property
     def headers(self):
         if not hasattr(self, '_headers'):
-            self._headers = dict(self.headers_tuple)
+            self._headers = dict()
```

Comment:
you could use `defaultdict(list)` here and decomplicate some of the `if k in d` logic.

Reply:
Great suggestion, changing now.